docs/spec_draft.md

```diff
@@ -2215,6 +2216,33 @@ implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.replica_id
+
+### Semantics
+
+Returns the unique id of the replica in the interval [0, N), where N is the
+number of replicas. Since all the replicas are running the same program, a
+`stablehlo.replica_id` call in the program will return a different value on each
+replica.
+
+### Outputs
+
+| Name     | Type                  |
+|----------|-----------------------|
+| `result` | tensor of type `ui32` |
```

Comment:
we can add the rank zero part here.
a 0  dimensional tensor
or a tensor constant (not sure it should be called a constant)

Reply:
We can then remove the constraints section altogether.

Reply:
+1